File: final/pb14.cpp
#include <iostream>
#include <vector>
using namespace std;


int main() {
  int n;
  cin >> n;
  vector<string> V(n);
  for (string& s : V) cin >> s;

  int m;
  cin >> m;
  while (m--) { // m vegades
    int p;
    char y;
    cin >> p >> y;
    --p;
    char x = V[0][p];
    int nx = 0;
    int ny = 0;
    for (const string& s : V)
      if (s[p] == x) ++nx;
      else if (s[p] == y) ++ny;
    cout << (2*ny <  nx ? "patologica" : "neutral") << endl;
  }
}

File: final/pb14.py
n = int(input())
matrix = [input() for i in range(n)]

m = int(input())
for i in range(m):
    p = int(input())
    c = input()

    nx = 0
    ny = 0
    for j in range(n):
        if matrix[j][p - 1] == matrix[0][p - 1]:
            nx = nx + 1
        if matrix[j][p - 1] == c:
            ny = ny + 1

    if 2 * ny >= nx:
        print('neutral')
    else:
        print('patologica')
